x/fswap/keeper/grpc_query_test.go

```diff
@@ -88,48 +89,229 @@ func (s *FSwapQueryTestSuite) SetupTest() {
-			}
+			response, err := s.queryClient.Swapped(s.ctx.Context(), tc.request)
+			s.Require().Equal(tc.expectedResponse, response)
+			actualGrpcCode := status.Code(err)
+			s.Require().Equal(tc.expectedGrpcCode, actualGrpcCode, actualGrpcCode.String())
+		})
+	}
+}
 
-			_, err := s.queryClient.Swap(s.ctx.Context(), m)
+func (s *FSwapQueryTestSuite) TestQueryTotalSwappableToCoinAmountRequest() {
+	tests := []struct {
+		name             string
+		request          *types.QueryTotalSwappableToCoinAmountRequest
+		expectedResponse *types.QueryTotalSwappableToCoinAmountResponse
+		expectedGrpcCode codes.Code
+	}{
+		{
+			name: "valid",
+			request: &types.QueryTotalSwappableToCoinAmountRequest{
+				FromDenom: s.fromDenom,
+				ToDenom:   s.toDenom,
+			},
+			expectedResponse: &types.QueryTotalSwappableToCoinAmountResponse{
+				SwappableAmount: sdk.NewCoin(s.toDenom, s.swap.AmountCapForToDenom),
+			},
+			expectedGrpcCode: codes.OK,
+		},
+		{
+			name: "invalid: empty fromDenom",
+			request: &types.QueryTotalSwappableToCoinAmountRequest{
+				FromDenom: "",
+				ToDenom:   s.toDenom,
+			},
+			expectedResponse: nil,
+			expectedGrpcCode: codes.Unknown,
+		},
+		{
+			name: "invalid: empty toDenom",
+			request: &types.QueryTotalSwappableToCoinAmountRequest{
+				FromDenom: s.fromDenom,
+				ToDenom:   "",
+			},
+			expectedResponse: nil,
+			expectedGrpcCode: codes.Unknown,
+		},
+		{
+			name: "invalid: unregistered swap",
+			request: &types.QueryTotalSwappableToCoinAmountRequest{
+				FromDenom: s.toDenom,
+				ToDenom:   s.fromDenom,
+			},
+			expectedResponse: nil,
+			expectedGrpcCode: codes.Unknown,
+		},
+	}
+	for _, tc := range tests {
+		s.Run(tc.name, func() {
+			response, err := s.queryClient.TotalSwappableToCoinAmount(s.ctx.Context(), tc.request)
+			s.Require().Equal(tc.expectedResponse, response)
+			actualGrpcCode := status.Code(err)
+			s.Require().Equal(tc.expectedGrpcCode, actualGrpcCode, actualGrpcCode.String())
+		})
+	}
+}
+
+func (s *FSwapQueryTestSuite) TestQuerySwapsRequest() {
+	tests := []struct {
+		name             string
+		request          *types.QuerySwapsRequest
+		expectedResponse *types.QuerySwapsResponse
+		expectedGrpcCode codes.Code
+	}{
+		{
+			name: "valid",
+			request: &types.QuerySwapsRequest{
+				Pagination: nil,
+			},
+			expectedResponse: &types.QuerySwapsResponse{
+				Swaps: []types.Swap{
+					{
+						FromDenom:           s.swap.FromDenom,
+						ToDenom:             s.swap.ToDenom,
+						AmountCapForToDenom: s.swap.AmountCapForToDenom,
+						SwapRate:            s.swap.SwapRate,
+					},
+				},
+				Pagination: &query.PageResponse{
+					NextKey: nil,
+					Total:   1,
+				},
+			},
+			expectedGrpcCode: codes.OK,
+		},
+		{
+			name: "invalid request",
+			request: &types.QuerySwapsRequest{
+				Pagination: &query.PageRequest{
+					Key:        []byte("invalid-key"),
+					Offset:     1,
+					Limit:      0,
+					CountTotal: false,
+					Reverse:    false,
```

Comment:
I think these two lines could be deleted. (Other test functions which test pagination haven't these options)